driver/src/test/java/org/neo4j/driver/internal/cluster/RediscoveryTest.java

```diff
@@ -454,10 +455,22 @@ private static ClusterComposition rediscover( BoltServerAddress initialRouter, C
         Clock mockedClock = mock( Clock.class );
         Logger mockedLogger = mock( Logger.class );
 
-        Rediscovery rediscovery = new Rediscovery( initialRouter, settings, mockedClock, mockedLogger, provider );
+        Rediscovery rediscovery = new Rediscovery( initialRouter, settings, mockedClock, mockedLogger, provider,
+                directMapProvider );
         return rediscovery.lookupClusterComposition( connections, routingTable );
     }
 
+    private static DnsResolver directMapProvider = new DnsResolver()
+    {
+        @Override
+        public Set<BoltServerAddress> resolve( BoltServerAddress initialRouter )
+        {
+            Set<BoltServerAddress> directMap = new HashSet<>( 1 );
```

Comment:
No, I cannot return any singlention set as it does not support `removeAll` method